src/object.d

```diff
@@ -3241,181 +3241,105 @@ template RTInfo(T)
     enum RTInfo = null;
 }
 
+// Compare class and interface objects for ordering.
+private int __cmp(Obj)(Obj lhs, Obj rhs)
+if (is(Obj : Object))
+{
+    if (lhs is rhs)
+        return 0;
+    // Regard null references as always being "less than"
+    if (!lhs)
+        return -1;
+    if (!rhs)
+        return 1;
+    return lhs.opCmp(rhs);
+}
+
+int __cmp(T)(const T[] lhs, const T[] rhs) @trusted
```

Comment:
@CyberShadow ok, so this overload should generate only one copy per scalar type. There are still a couple of opportunities:

* merge wchar with short implementation
* merge dchar with uint implementation
* merge ifloat with float implementation
* merge idouble with real implementation
* merge ireal with real implementation

I've let these be because they're relatively rare.